### PR TITLE
[WIP] try out minIO

### PR DESCRIPTION
**Pull request by chGoodchild**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -66,18 +66,54 @@ jobs:
           ninja
           ninja test
 
+      - name: Set up Go
```
Comment (wolfv) [inline .github/workflows/main.yml:69]: This shouldn't be necessary for us

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -66,18 +66,54 @@ jobs:
...
+                     -v /tmp/config:/root/.minio \
+                     minio/minio server /data
+
+          sudo docker start minio
+
+          export AWS_ACCESS_KEY_ID=minioadmin
+          export AWS_SECRET_ACCESS_KEY=minioadmin
+          export AWS_EC2_METADATA_DISABLED=true
+
+          # pip3 install awscli --force-reinstall --upgrade
```
Comment (wolfv) [inline .github/workflows/main.yml:89]: this you probably need to have the `aws` command.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -66,18 +66,54 @@ jobs:
...
+          # Download the packages from conda forge
           server_path="test/conda_mock/static/packages"
           powerloader_path="build/powerloader"
           wget https://beta.mamba.pm/get/conda-forge/osx-arm64/xtensor-0.24.0-hc021e02_0.tar.bz2 \
               -O ./test/conda_mock/static/packages/xtensor-0.24.0-hc021e02_0.tar.bz2
           $powerloader_path download -f test/remote_mirrors.yml -d $server_path
+
+          # Populate minIO
+          # https://docs.min.io/docs/aws-cli-with-minio.html
+          aws --endpoint-url $LOCAL_AWS_S3_ENDPOINT s3 mb s3://testbucket
```
Comment (wolfv) [inline .github/workflows/main.yml:114]: this doesn't work if you don't install the `awscli`

Comment (chGoodchild) [reply .github/workflows/main.yml:114]: I installed it manually on my machine and then added it to `environment.yml`

File: .github/workflows/main.yml (hunk 1 above)
Comment (chGoodchild) [reply .github/workflows/main.yml:69]: true..

File: .github/workflows/main.yml (hunk 3 above)
Comment (wolfv) [reply .github/workflows/main.yml:109]: makes sense

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -66,6 +72,24 @@ jobs:
...
+                     -v /tmp/config:/root/.minio \
+                     minio/minio server /data
+
+          export AWS_ACCESS_KEY_ID=minioadmin
+          export AWS_SECRET_ACCESS_KEY=minioadmin
+          export AWS_EC2_METADATA_DISABLED=true
+
+          # Populate minIO
+          # https://docs.min.io/docs/aws-cli-with-minio.html
+          sudo docker start minio
```
Comment (wolfv) [inline .github/workflows/main.yml:90]: isn't it already running from above?

**Comment by chGoodchild** [comment]
Started a new branch for this: https://github.com/chGoodchild/powerloader/tree/mock_s3_clean_slate

**Pull request closed without merging**